### ISSUE TITLE
Comments in commit details are not ignored

### ISSUE THREAD

**Issue by Turbo87** [ForkIssues/Tracker#180]
In most other git apps lines in the commit message starting with `#` are ignored for the final commit message. It seems that Fork is including those lines, which can be problematic if the commit message was automatically generated by git outside of Fork.

**Comment by DanPristupov** [comment]
Seeing the `#` lines is absolutely fine for me and also helpful for mentioning conflicts for merge commits. What particular cases do you find problematic?

**Comment by Turbo87** [comment]
The issue is that this is non-standard behavior. I'm okay with seeing those lines while entering the commit message, but they should not show up in the final commit message. It's causing things to be included in the message that were not intended to be included.

**Comment by Brunty** [comment]
I have an issue related to this:

I have a commit template that has lines starting with `#` - it's used as a reminder on how to format my commits and when working with `git commit` on the CLI it populates VIM, but as the lines start with `#` they're ignored as they're comments.

With Fork, it adds this template to the commit description and the lines aren't ignored, so commits get filled with the template content.

**Comment by DanPristupov** [comment]
This is the default behavior.

Does changing the `commit.cleanup` config value help?

`$ git config --local commit.cleanup strip`

P.S. Command to undo the change: `$ git config --local --unset commit.cleanup`.

**Comment by svvitale** [comment]
That did it, thank you!

**Comment by olets** [comment]
> `git config --local commit.cleanup strip`

works but

> This is the default behavior.

is not my experience. When I compose a message in the editor (ie `git commit [--amend]` without `-m`) I get this note in `.git/COMMIT_EDITMSG`:

```shell
# Please enter the commit message for your changes. Lines starting
# with '#' will be ignored, and an empty message aborts the commit.
```

That applies to that default comment, any comments added while writing the commit, merge conflict file lists, and comments in the commit template (`git config commit.template`). Like @Brunty my template includes comments, so this behavior discourages me from committing via Fork — I have to make sure to delete the comments.

> helpful for mentioning conflicts for merge commits

Fork's showing the conflict list in the UI is fine, but preserving those comments in the final commit message is a pain point for me. My collaborators and I don't need to know that I resolved conflicts while rewriting history. As with template comments I can delete them of course, but I have to remember to and then have to take the extra action.

**Comment by DanPristupov** [comment]
As for today, I disagree with my comment above https://github.com/fork-dev/Tracker/issues/180#issuecomment-352252040. I was incorrectly sure this is how it works by default.

The problem is more complicated than it seems though.

1. GitHub and other git services  recognize the `#` symbol as a reference to issue/PR (e.g. #180). As a result, many teams use a rule to prefix each commit with `#xxx` to mark the issue/PR number.
2. 'Commit' with leading `#` in subject works fine, but when users try to rebase such commit, the operation fails because `#` is the comment symbol.
3. Fork solves this by overriding `core.commentChar=^`. I admin, it's not nice, but it works quite well. Nobody reported a problem with rebase since I changed it.

What can be a possible solution?
- Enabling `commit.cleanup strip` won't help in your case because Fork uses `^` as the comment symbol
- We can't return to `#` because a lot of people will have problem with rebase

**Comment by olets** [comment]
Stopping by to say this remains my only frustration with Fork. Imo (2) is user error and (3) is not Fork's responsibility. (3) also enables a practice that will break when not using Fork. Respecting the repo commentChar config (and possibly including a release note about how to configure Git to support initial `#`) would satisfy this (and #203), and would protect users who use initial `#` from having trouble when not using Fork. This comes up probably at least every week for me, but usually more frequently; it isn't unusual for it to get me multiple times a day as I juggle multiple overlapping but independent features.

**Comment by cyanite** [comment]
@DanPristupov If Fork redefines the comment char, it should pre-strip any commit message from Git starting with "#". Wouldn't that work?

**Comment by saagarjha** [comment]
Not sure if anyone has mentioned this above, but this breaks the rebase workflow when there are conflicts. In particular, when a rebase is running and conflicts occur, then git will create a commit message that looks like this:

```
[your original commit message]

# Conflicts:
#	[conflicting files]
```

Without Fork, I resolve the conflicts, run `git rebase --continue`, and the commented lines get dropped with the commit message remaining as-is. If I hit "continue rebase" in Fork though the commented lines with the conflicts list gets added to the commit, which is never what I want. What's more, Fork disables the commit message field when resolving a conflict during a rebase, so I don't even see a way to remove this line during the rebase. This means I can never continue a rebase in Fork if there are conflicts, which is quite annoying :(

**Comment by lobsterkatie** [comment]
> What's more, Fork disables the commit message field when resolving a conflict during a rebase, so I don't even see a way to remove this line during the rebase.

Yes, this is my problem, too. It'd be annoying to have to remove the `# conflicts: ...` lines every time, but at least if that field were editable, I'd be able to do so. As it is, I have to finish the rebase, then go back and do _another_ rebase to "reword" every commit message that got tagged with conflicts (i.e., delete the extra lines).

> Fork solves this by overriding core.commentChar=^

Here's an interesting thing I've noticed. The conflicts _do_ seem to show up with `^` (rather than a `#`) if I'm doing a cherry-pick, and they are indeed correctly ignored as comments and not included in the final commit message. But if I'm doing a rebase, they show up with `#` and end up polluting my commit messages. So could rebase just switch to using the same behavior as cherry-pick? Seems like that'd solve the problem for everyone.

**Comment by polw-zi** [comment]
Seconding the annoyance of rebases now including this message with no way to remove them.

Given that Fork already messes with the comment character, I wonder if it'd be acceptable for Fork to do its own comment stripping, but based on a prefix that includes a space after the hash (`"# "`).
